### PR TITLE
feat(lwm2m): always publish update message

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
Fixes https://github.com/emqx/emqx-lwm2m/issues/90

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked.:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

Add configurations to always publish messages for UPDATE LwM2M messages.

### CODE REVIEW

File: apps/emqx_lwm2m/priv/emqx_lwm2m.schema [hunk 1]
```diff
@@ -112,6 +112,11 @@ end}.
   {default, "lwm2m/%e/up/resp"}
 ]}.
 
+{mapping, "lwm2m.publish_update_when", "emqx_lwm2m.publish_update_when", [
+  {datatype, atom},
```
Comment (zmstone) [inline apps/emqx_lwm2m/priv/emqx_lwm2m.schema:116]: `enum` ?

File: apps/emqx_lwm2m/src/emqx_lwm2m_protocol.erl [hunk 2]
```diff
@@ -115,15 +115,23 @@ post_init(Lwm2mState = #lwm2m_state{endpoint_name = _EndpointName,
...
 
-    %% - report the registration info update, but only when objectList is updated.
-    case NewRegInfo of
-        #{<<"objectList">> := _} ->
+    case proplists:get_value(publish_update_when,
+            lwm2m_coap_responder:options(), object_list_changed) of
+        always ->
             send_to_broker(<<"update">>, #{<<"data">> => UpdatedRegInfo}, Lwm2mState);
-        _ -> ok
+        _ ->
```
Comment (zmstone) [inline apps/emqx_lwm2m/src/emqx_lwm2m_protocol.erl:128]:
```suggestion
        object_list_changed ->
```

File: apps/emqx_lwm2m/etc/emqx_lwm2m.conf [hunk 3]
```diff
@@ -41,6 +41,11 @@ lwm2m.topics.register = up/resp
 # The topic to which the lwm2m client's update message is published
 lwm2m.topics.update = up/resp
 
+# Always publish the update message only if the object list is changed.
```
Comment (zmstone) [inline apps/emqx_lwm2m/etc/emqx_lwm2m.conf:44]:
```suggestion
# Configure when to publish update message.
```

**Review by zmstone — COMMENTED** [review]
Please link a PR to https://github.com/emqx/emqx-docs

File: apps/emqx_lwm2m/priv/emqx_lwm2m.schema (hunk 1 above)
Comment (terry-xiaoyu) [reply apps/emqx_lwm2m/priv/emqx_lwm2m.schema:116]: Yes, it should be of type enum

File: apps/emqx_lwm2m/etc/emqx_lwm2m.conf (hunk 3 above)
Comment (terry-xiaoyu) [reply apps/emqx_lwm2m/etc/emqx_lwm2m.conf:44]: I'll change it

File: apps/emqx_lwm2m/src/emqx_lwm2m_protocol.erl (hunk 2 above)
Comment (terry-xiaoyu) [reply apps/emqx_lwm2m/src/emqx_lwm2m_protocol.erl:128]: OK

**Review by zmstone — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Pull request merged**